src/Compilers/CSharp/Test/Semantic/Semantics/AccessCheckTests.cs

```diff
@@ -753,34 +742,209 @@ class ADerived2: A
             FieldSymbol protField = classA.GetMembers("prot").Single() as FieldSymbol;
             TypeSymbol karrayType = (classA.GetMembers("karray").Single() as FieldSymbol).Type;
             TypeSymbol aarrayType = (classA.GetMembers("aarray").Single() as FieldSymbol).Type;
+            TypeSymbol kptrType = (classA.GetMembers("kptr").Single() as FieldSymbol).Type;
+            TypeSymbol aptrType = (classA.GetMembers("aptr").Single() as FieldSymbol).Type;
             TypeSymbol kenumType = (classA.GetMembers("kenum").Single() as FieldSymbol).Type;
             TypeSymbol aenumType = (classA.GetMembers("aenum").Single() as FieldSymbol).Type;
+            var discards = tree.GetRoot().DescendantNodes().OfType<IdentifierNameSyntax>().Where(i => i.Identifier.ContextualKind() == SyntaxKind.UnderscoreToken).ToArray();
+            DiscardSymbol kdiscard = (DiscardSymbol)model.GetSymbolInfo(discards[0]).Symbol;
+            DiscardSymbol adiscard = (DiscardSymbol)model.GetSymbolInfo(discards[1]).Symbol;
             TypeSymbol unknownType = (classA.GetMembers("unknowntype").Single() as FieldSymbol).Type;
-            var semanticModel = c.GetSemanticModel(c.SyntaxTrees[0]);
+
+            ISymbol nullSymbol = null;
+            Assert.Throws<ArgumentNullException>(() => { compilation.IsSymbolAccessibleWithin(classA, nullSymbol); });
```

Comment:
Consider breaking input validation into a separate test.

Reply:
I'd rather keep it here since it benefits from the setup that defines the locals used.

---
In reply to: [202466419](https://github.com/dotnet/roslyn/pull/27312#discussion_r202466419) [](ancestors = 202466419)